src/general/WorldGenerationSettings.cs

```diff
@@ -124,4 +134,48 @@ public override string ToString()
             $", Easter eggs: {EasterEggs}" +
             "]";
     }
+
+    /// <summary>
+    ///   Generates a formatted string containing translated difficulty details.
+    /// </summary>
+    public string GetTranslatedDifficultyString()
+    {
+        string translatedDifficulty = Difficulty is DifficultyPreset difficulty ?
+            difficulty.Name :
+            TranslationServer.Translate("DIFFICULTY_PRESET_CUSTOM");
+
+        return string.Format(CultureInfo.CurrentCulture, TranslationServer.Translate("DIFFICULTY_DETAILS_STRING"),
+            translatedDifficulty,
+            MPMultiplier,
+            AIMutationMultiplier,
+            CompoundDensity,
+            PlayerDeathPopulationPenalty,
+            TranslationServer.Translate("PERCENTAGE_VALUE").FormatSafe(Math.Round(GlucoseDecay * 100, 1)),
+            OsmoregulationMultiplier,
+            TranslationHelper.TranslateBool(FreeGlucoseCloud),
+            TranslationHelper.TranslateBool(PassiveGainOfReproductionCompounds),
+            TranslationHelper.TranslateBool(LimitReproductionCompoundUseSpeed));
+    }
+
+    /// <summary>
+    ///   Generates a formatted string containing translated planet details.
+    /// </summary>
+    public string GetTranslatedPlanetString()
+    {
+        return string.Format(CultureInfo.CurrentCulture, TranslationServer.Translate("PLANET_DETAILS_STRING"),
+            TranslationServer.Translate(MapType.GetAttribute<DescriptionAttribute>()?.Description),
```

Comment:
Doesn't this pretty much completely fail to fulfill the purpose of this method if the getattribute fails? I think a cleaner approach would be to have translation helper take in a generic enum value (if possible) and if not possible then here using an attribute get overload that throws on failure to find would be at least a bit better.

Reply:
I'm struggling to parse exactly what you mean here. I agree it would be better to have an approach that throws if no description is found but I can't figure out what your specific suggestion is.

Reply:
Modify `GetAttribute` to return `T` and throw if it can't find the value.

Actually it looks like the method exists already before this PR so it might be a bit tricky to modify it depending on where it is used elsewhere, but I'd really like it to look like this:
```
    public static T GetAttribute<T>(this Enum e)
        where T : Attribute
    {
        return GetAttributes<T>(e).First();
    }
```

Reply:
The only other place this is used throws if it returns null anyway, so this should be fine.